### PR TITLE
feat: cache `auth` in RSC

### PR DESCRIPTION
**Pull request by dBianchii**
This PR changes the `auth()` export in @acme/api to go through react/cache. This will allow for de-duplicating of session calls and will only do one per request across all components

### CODE REVIEW

**Comment by dBianchii** [comment]
I wonder if we should do an example of calling auth() in 2 separate RSCs, to showcase how you don't need to pass session around every time

**Comment by dBianchii** [comment]
@juliusmarminge after we approve this, let me know if I can do the same over on create-t3-app. I think it's also not using react/cache with the app router

File: packages/auth/src/index.ts
Comment (juliusmarminge) [inline packages/auth/src/index.ts:47]: This doesn't work in middleware does it?

Comment (dBianchii) [reply packages/auth/src/index.ts:47]: In Next's middleware.ts?

Comment (dBianchii) [reply packages/auth/src/index.ts:48]: Then I believe we could add a comment somewhere, or in Readme.md explaining to also export `defaultAuth`  for that use case. What do you suggest?

Comment (juliusmarminge) [reply packages/auth/src/index.ts:48]: i think we should have a middleware.ts file by default tbh

Comment (dBianchii) [reply packages/auth/src/index.ts:48]: ¯\_(ツ)_/¯ in this case yea sure

Just thought that this repo was supposed to be a "turbo" version of ct3a. And since in there we don't specify middleware files, don't see why we should also do it here? Could be a good idea though.

Comment (juliusmarminge) [reply packages/auth/src/index.ts:48]: afaik middleware is the only way to make the session refresh?

**Review by juliusmarminge — APPROVED** [review]

**Comment by dBianchii** [comment]
Wait, but with this setup how would you import from index.rsc? I see that `auth-showcase.tsx` is still efectively importing from index.tsx only

File: packages/auth/package.json
Comment (juliusmarminge) [inline packages/auth/package.json:8]: This here ^^

**Comment by dBianchii** [comment]
> Exports condition

Cntrl + click on auth-showcase.ts auth() still is bringing me to index.ts tho, not index.rsc.ts

**Comment by juliusmarminge** [comment]
yea the default entrypoint is still the normal index file.

exports condition are evaluated at runtime based on the thing that imports it

**Comment by juliusmarminge** [comment]
See https://github.com/reactjs/rfcs/blob/main/text/0227-server-module-conventions.md#react-server-conditional-exports

**Pull request merged**
